spec/features/users_can_create_a_project_level_activity_spec.rb

```diff
@@ -192,6 +192,43 @@
         expect(created_activity.implementing_organisations).to be_none
       end
 
+      scenario "a non-ODA project can be linked to an existing ODA project" do
+        oda_programme = create(:programme_activity, :ispf_funded,
+          is_oda: true,
+          extending_organisation: user.organisation)
+        _report = create(:report, :active, organisation: user.organisation, fund: oda_programme.associated_fund)
+        oda_project = create(:project_activity, :ispf_funded,
+          parent: oda_programme,
+          is_oda: true,
+          ispf_theme: 1)
+
+        non_oda_programme = create(:programme_activity, :ispf_funded,
+          is_oda: false,
+          linked_activity: oda_programme,
+          extending_organisation: user.organisation)
+        non_oda_project = build(:project_activity, :ispf_funded,
+          parent: non_oda_programme,
+          is_oda: false,
+          linked_activity_id: oda_project.id,
```

Comment:
Yes, we need to set the attribute rather than the association when the object isn't persisted. The only reason we have `non_oda_project` is to copy the attributes off of it in the `ActivityForm`, and we want to be able to copy `linked_activity_id`, but we don't want the association to be saved, because it would make the linked activity unavailable (and other attributes non-unique).

Reply:
Ah yes, didn't spot this is `build`!